File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i =0
        j = len(numbers)-1
        suma = numbers[i] + numbers[j]

        while suma != target:
            if target < suma:
                j -= 1
            else:
                i += 1
            suma = numbers[i] + numbers[j] 
        
        return [i+1, j+1]

File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    void help_funcion (vector<int>& nums, int target, vector<int>& ans, int sum, int i, vector<vector<int>>& endgame)
    {
        if(i >= nums.size())return;
        if(sum > target)return;
        if(sum == target)
        {
            endgame.push_back(ans);
            return;
        }
        ans.push_back(nums[i]);
        help_funcion(nums, target, ans, sum+nums[i], i, endgame);
        ans.pop_back();
        help_funcion(nums, target, ans, sum, i+1, endgame);
    }

    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<vector<int>> endgame{};
        vector<int>ans{};
        help_funcion(nums, target, ans, 0, 0, endgame);
        return endgame;
    }
};
